validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java

```diff
@@ -299,8 +304,36 @@ public void subscribeToPersistentSubnets(final Set<SubnetSubscription> subnetSub
                         s.getSubnetId(), s.getUnsubscriptionSlot()))
             .collect(Collectors.toSet());
 
-    asyncRunner
-        .runAsync(() -> apiClient.subscribeToPersistentSubnets(schemaSubscriptions))
+    sendRequest(() -> apiClient.subscribeToPersistentSubnets(schemaSubscriptions))
         .finish(error -> LOG.error("Failed to subscribe to persistent subnets", error));
   }
+
+  private SafeFuture<Void> sendRequest(final ExceptionThrowingRunnable requestExecutor) {
+    return sendRequest(
+        () -> {
+          requestExecutor.run();
+          return null;
+        });
+  }
+
+  private <T> SafeFuture<T> sendRequest(final ExceptionThrowingSupplier<T> requestExecutor) {
+    return asyncRunner.runAsync(() -> sendRequest(requestExecutor, 0));
+  }
+
+  private <T> SafeFuture<T> sendRequest(
+      final ExceptionThrowingSupplier<T> requestExecutor, final int attempt) {
+    return SafeFuture.of(requestExecutor)
+        .exceptionallyCompose(
+            error -> {
+              if (Throwables.getRootCause(error) instanceof RateLimitedException
+                  && attempt < MAX_RATE_LIMITING_RETRIES) {
+                LOG.warn(
+                    "Received Too Many Requests response from beacon node. Retrying after a delay.");
```

Comment:
`Beacon node has warned that we're making too many requests. Retrying after delay.`